cs130_front/src/components/views/ClassesPage/ClassesPage.js

```diff
@@ -11,42 +12,155 @@ import SearchBar from '../../Search/SearchBar';
+    }
+  }
+  
+  useEffect(() => {
+		async function initCourses() {
+			try {
+        const userResponse = await getUser();
+        const allGroups = handleUserResponse(userResponse.data.groups);
+        const courseResponse = await getCourses();
+        handleBuddiesListingsResponse(courseResponse.data, allGroups);
+        const g = await getUsers();
+			} catch (err) {
+				// Handle err here. Either ignore the error, or surface the error up to the user somehow.
+			}
+		}
+    initCourses();
+  }, []);
+
+  function getCourses() {
+    return axios.get(`${USER_SERVER_AUTH}/getBuddiesListings?u_id=${userId}`, config);
+  }
+
+  function getUser() {
+    return axios.get(`${USER_SERVER_AUTH}?u_id=${userId}`, config);
+  }
+
+  function getUsers () {
+    return axios.get(`${ROUTE}/getAllUsers`);
+  }
+
+  function handleUserResponse(data) {
+    const allGroups = {};
+    Object.keys(data).forEach(function(key) {
+      const name = data[key]["name"];
+      const groupId = data[key]["id"];
+      const currentGroup = {name: name, groupId: groupId};
+      if (!(data[key]["course_name"] in allGroups)){
+        allGroups[data[key]["course_name"]] = [];
+      }
+      allGroups[data[key]["course_name"]].push(currentGroup);
+    });
+    setAllGroups(allGroups);
+    return allGroups;
+  }
+
+  function handleBuddiesListingsResponse(data, allGroups){
+
+    const i = [{name: "Wow Squad", groupId: 123}, {name: "CA Squad", groupId: 124}, {name: "DM Squad", groupId: 125}]
+
+    const classes = []
+    Object.keys(data).forEach(function(key) {
+      const studyBuddies = [];
+      const userDict = {};
+      Object.keys(data[key]["StudyBuddies"]).forEach(function(key2) {
+        const name = data[key]["StudyBuddies"][key2]["first_name"] + " " + data[key]["StudyBuddies"][key2]["last_name"];
+        const id = data[key]["StudyBuddies"][key2]["u_id"];
+        const school = data[key]["StudyBuddies"][key2]["school_name"];
+        const email = data[key]["StudyBuddies"][key2]["u_email"];
+        const discord = "None";
+        if (!(id == parseInt(userId))){
+          studyBuddies.push({name: name, school: school, id: id, discord: discord, email: email});
+        }
+        userDict[id] = {name: name, school: school};
+      });
+
+      const courseId = key;
+      const name = data[key]["CourseName"];
+      const listings = [];
+      console.log(data[key]["Listings"]);
+      Object.keys(data[key]["Listings"]).forEach(function(key2) {
+        const poster = data[key]["Listings"][key2]["poster"];
+        const name = userDict[poster].name;
+        const school = userDict[data[key]["Listings"][key2]["poster"]].school;
+        const description = data[key]["Listings"][key2]["text_description"];
+        const group_id = data[key]["Listings"][key2]["group_id"];
+        const group_name = data[key]["Listings"][key2]["group_name"];
+        if (!(data[key]["Listings"][key2]["poster"] == parseInt(userId))){
```

Comment:
i think a user should be able to see their listings. I would remove this if condition.